docs/filters/filter.lua

```diff
@@ -0,0 +1,106 @@
+--[[
+function dump(o)
+   if type(o) == 'table' then
+      local s = '{ '
+      for k,v in pairs(o) do
+         if type(k) ~= 'number' then k = '"'..k..'"' end
+         s = s .. '['..k..'] = ' .. dump(v) .. ','
+      end
+      return s .. '} '
+   else
+      return tostring(o)
+   end
+end
+]]--
+
+-- uppercase top-level headings for man pages
+function Header(el)
+  if FORMAT == "man" and el.level == 1 then
+    return el:walk{Str = function(el) return pandoc.Str(string.upper(el.text)) end}
+  end
+end
+
+-- inline code is assumed to be in jq
+function Code(code)
+  code.classes[1] = "jq"
+  return code
+end
+
+-- code blocks are assumed to be in jq if no other language is given
+function CodeBlock(block)
+  if next(block.classes) == nil then
+    block.classes[1] = "jq"
+    return block
+  end
+end
+
+function Div(el)
+  local class = el.classes[1]
+  if class == "Examples" then
+    el = el:walk{CodeBlock = function(block) return example(block.text) end}
+
+    if FORMAT == "html" then
+      local summary = pandoc.Plain{
+        pandoc.RawInline("html",  "<summary>"), pandoc.Str 'Examples',
+        pandoc.RawInline("html", "</summary>")
+      }
+      return {
+        pandoc.RawBlock("html",  '<details class="pb-3">'), summary, el,
+        pandoc.RawBlock("html", '</details>')
+      }
+    end
+  end
+  if FORMAT == "html" then
+    el.classes = {"alert", "alert-primary"}
+    table.insert(el.content, 1, pandoc.Span(class, {class = "block-start"}))
+  else
+    el = pandoc.DefinitionList({{pandoc.Emph(class), el.content}})
+  end
+  return el
+end
+
+function example(test)
+  local _, _, filter, input, output = test:find("([^\n]+)\n([^\n]+)\n(.*)")
+
+  if FORMAT == "man" then
+    filter = "'" .. filter .. "'"
+    input = "'" .. input .. "'"
+    return pandoc.CodeBlock("$ jq " .. filter .. " \\\n  <<< " .. input .. "\n" .. output)
```

Comment:
> For the web version multiple outputs seems to end up on same line

This is now corrected in 52ce7951cfc0d122a800177af77d9029c027cd29.

Reply:
> Maybe example sections should have background color? same as note/compatibility?

Example sections are different, because they can be quite long and they are not shown by default. For that reason, I would not color them.